README.md

```diff
@@ -1,138 +1,664 @@
+
+Despite these differences, it should generally be possible to achieve the same sort of functionality with this decorators proposal as with Babel legacy decorators. If you see important missing functionality in this proposal, please file an issue.
+
+#### Comparison with TypeScript "experimental" decorators
+
+TypeScript experimental decorators are largely similar to Babel legacy decorators, so the comments in that section apply as well. In addition:
+- This proposal does not include parameter decorators, but they may be provided by future built-in decorators, see [NEXTBUILTINS.md](./NEXTBUILTINS.md).
+- TypeScript decorators run all instance decorators before all static decorators, whereas the order of evaluation (for both `@wrap` and `@register`) in this proposal is based on the ordering in the program, regardless of whether they are static or instance.
+
+Despite these differences, it should generally be possible to achieve the same sort of functionality with this decorators proposal as with TypeScript experimental decorators. If you see important missing functionality in this proposal, please file an issue.
+
+#### Comparison with the previous Stage 2 decorators proposal
+
+The previous Stage 2 decorators proposal was more full-featured than this proposal, including:
+- Access to reading and writing private fields, and declaring new private fields
+- Class decorator access to manipulating all fields and methods within the class
+- More flexible handling of the initializer, treating it as a "thunk"
+
+These features aren't included in this initial proposal, but they may be provided by future built-in decorators.
+
+The previous Stage 2 decorators proposal was based on a concept of descriptors which stand in for various class elements. Such descriptors do not exist in this proposal, but could be partially revived in future built-in decorators. However, those descriptors gave a bit too much flexibility/dynamism to the class shape in order to be efficiently optimizable; future built-in decorators would add the same functionality in a more statically analyzable way.
+
+### If the Stage 2 decorators didn't work out, why not go back and standardize TS/Babel legacy decorators?
+
+**Path towards features and analyzability**: Legacy decorators are run as a function, and they don't give any clear path towards being statically analyzable or expandable ahead of the time through tools, or a way to extend them to other possibilities, such as decorating field initializers, private class elements, functions, objects, etc.
+
+**Technical infeasibility**: Legacy decorators, when applied to field declarations, depend deeply on the semantics that field initializers call setters. TC39 [concluded](https://github.com/tc39/proposal-class-fields/blob/master/README.md#public-fields-created-with-objectdefineproperty) that, instead, field declarations act like Object.defineProperty. This decision makes many patterns with legacy decorators no longer work.
```

Comment:
Quick note on this, it seems like TS decorators rely on this, but Babel Stage 1 (legacy) decorators _do_ use `Object.defineProperty()` for any decorated field. Strangely enough, they require loose mode for class fields currently, but don't apply the same constraint to decorated fields. This means you can't for instance decorate a field into a getter, and assign its value with the field initializer.

We're planning on using a custom babel transform for certain decorators to move field initializers into the constructor to get around this, essentially making them work like TS decorators, but figured it would be good to mention this somewhere.

Reply:
Interesting, do you have a link I can include that describes the approach?

Reply:
Making a note that I will as soon as we get it set up!